support/azureutil/azureutil.go

```diff
@@ -0,0 +1,118 @@
+	return subnet.Name, nil
+}
+
+// GetNetworkSecurityGroupNameFromNetworkSecurityGroupID extracts the network security group (nsg) name from a nsg ID
+// Example nsg ID: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/networkSecurityGroups/<nsgName>
+func GetNetworkSecurityGroupNameFromNetworkSecurityGroupID(nsgID string) (string, error) {
+	nsg, err := arm.ParseResourceID(nsgID)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse network security group ID %q: %v", nsgID, err)
+	}
+
+	if nsg.Name == "" {
+		return "", fmt.Errorf("failed to parse network security group name from %q", nsgID)
+	}
+
+	return nsg.Name, nil
+}
+
+// GetVnetNameAndResourceGroupFromVnetID extracts the VNET name and its resource group from a VNET ID
+// Example VNET ID: /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>
+func GetVnetNameAndResourceGroupFromVnetID(vnetID string) (string, string, error) {
+	vnet, err := arm.ParseResourceID(vnetID)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to parse vnet ID %q: %v", vnetID, err)
+	}
+
+	if vnet.Name == "" {
+		return "", "", fmt.Errorf("failed to parse vnet name from %q", vnetID)
+	}
+
+	if vnet.ResourceGroupName == "" {
+		return "", "", fmt.Errorf("failed to parse vnet resource group name from %q", vnetID)
+	}
+
+	return vnet.Name, vnet.ResourceGroupName, nil
+}
+
+// GetVnetInfoFromVnetID extracts the full information on a VNET from a VNET ID by first getting the VNET name and
+// its resource group's name and then using those parameters to query the full information on the VNET using Azure's SDK
+func GetVnetInfoFromVnetID(ctx context.Context, vnetID string, subscriptionID string, azureCreds azcore.TokenCredential) (armnetwork.VirtualNetworksClientGetResponse, error) {
+	partialVnetInfo, err := arm.ParseResourceID(vnetID)
+	if err != nil {
+		return armnetwork.VirtualNetworksClientGetResponse{}, fmt.Errorf("failed to parse vnet information from vnet ID %q: %v", vnetID, err)
+	}
+
+	if partialVnetInfo.Name == "" {
+		return armnetwork.VirtualNetworksClientGetResponse{}, fmt.Errorf("failed to parse vnet name from %q", vnetID)
+	}
+
+	if partialVnetInfo.ResourceGroupName == "" {
+		return armnetwork.VirtualNetworksClientGetResponse{}, fmt.Errorf("failed to parse vnet resource group name from %q", vnetID)
+	}
+
+	vnet, err := getFullVnetInfo(ctx, subscriptionID, partialVnetInfo.ResourceGroupName, partialVnetInfo.Name, azureCreds)
+	if err != nil {
+		return armnetwork.VirtualNetworksClientGetResponse{}, err
+	}
+
+	return vnet, nil
+}
+
+// getFullVnetInfo gets the full information on a VNET
+func getFullVnetInfo(ctx context.Context, subscriptionID string, vnetResourceGroupName string, clientVnetName string, azureCreds azcore.TokenCredential) (armnetwork.VirtualNetworksClientGetResponse, error) {
+	networksClient, err := armnetwork.NewVirtualNetworksClient(subscriptionID, azureCreds, nil)
```

Comment:
It looks like there are a few places this needs to be fixed.  Since it's increasing scope to fix them all going to make a bug.  Feel free to proceed how you want here.

Reply:
>it looks like NewVirtualNetworksClient is hardcoded to public cloud. You'll have to do something to support govcloud in the future, or now.

Discussed this with Ben. He will create a bug for us to pass the cloud in as in option.

Reply:
[OCPBUGS-33372](https://issues.redhat.com/browse/OCPBUGS-33372)